alibi_detect/cd/tensorflow/mmd_online.py

```diff
@@ -168,7 +168,7 @@ def _configure_thresholds(self):
                     2 * tf.reduce_sum(k_xy_col_sums[w:w + w_size]))
                     for k_xx_sum, y_inds_w, k_xy_col_sums in zip(k_xx_sums_all, y_inds_all_w, k_xy_col_sums_all)
                     ]
-            mmds = tf.concat(mmds, axis=0)  # an mmd for each bootstrap sample
+            mmds = tf.stack(mmds, axis=0)  # an mmd for each bootstrap sample
```

Comment:
To fix the following error:

```python
Traceback (most recent call last):
  File "/home/runner/work/alibi-detect/alibi-detect/alibi_detect/cd/tests/test_mmd_online.py", line 25, in test_mmddriftonline
    cd = MMDDriftOnline(x_ref=x_ref, ert=25, window_size=5, backend=backend, n_bootstraps=100)
  File "/home/runner/work/alibi-detect/alibi-detect/alibi_detect/cd/mmd_online.py", line 99, in __init__
    self._detector = MMDDriftOnlineTF(*args, **kwargs)  # type: ignore
  File "/home/runner/work/alibi-detect/alibi-detect/alibi_detect/cd/tensorflow/mmd_online.py", line 87, in __init__
    self._configure_thresholds()
  File "/home/runner/work/alibi-detect/alibi-detect/alibi_detect/cd/tensorflow/mmd_online.py", line 171, in _configure_thresholds
    mmds = tf.concat(mmds, axis=0)  # an mmd for each bootstrap sample
  File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/tensorflow/python/util/traceback_utils.py", line [153](https://github.com/SeldonIO/alibi-detect/pull/739/checks#step:8:154), in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/tensorflow/python/framework/ops.py", line 7215, in raise_from_not_ok_status
    raise core._status_to_exception(e) from None  # pylint: disable=protected-access
tensorflow.python.framework.errors_impl.InvalidArgumentError: {{function_node __wrapped__ConcatV2_N_100_device_/job:localhost/replica:0/task:0/device:CPU:0}} ConcatOp : Can't concatenate scalars (use tf.stack instead) [Op:ConcatV2] name: concat
```

This has only begun occuring in tf 2.11. Prior to reaching the `tf.concat` call, `mmds` is a list of `tf.Tensor`'s each with `shape=()` (i.e. scalars!). The error message "_ConcatOp : Can't concatenate scalars_" therefore makes some sense. However, I cannot find anything in the 2.11 release notes about the behavior of `tf.concat` changing...

Reply:
Yes, at least for `tensorflow >= 2.2.0` which is the lower bound we support. The difference between the two is that `tf.concat` only works when `axis` is within the existing dimensions, whilst `tf.stack` creates a new dimension when `axis` is outside the existing dimensions. I think `stack` makes more sense here since we're concatenating scalars with zero dimensions into a tensor with dimension `(n,)`. In other words, it seems that 2.11 has simply made `tf.concat` more strict about "misuse", rather than it changing the behavior of `stack` or `concat`.